src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -213,7 +213,7 @@ internal static void IntrospectiveSort(T[] keys, int left, int length, Compariso
             if (length < 2)
                 return;
 
-            IntroSort(keys, left, length + left - 1, 2 * IntrospectiveSortUtilities.FloorLog2PlusOne(keys.Length), comparer);
```

Comment:
The same bug seems to be in the non-generic sort version in Array.cs. Could you please fix it there as well?

Reply:
@jkotas done, hope I've got them all now :)